### DISCUSSION TITLE
Infinity Handling

### DISCUSSION THREAD

**Discussion by KevinAlistairMurray** [exceljs/exceljs#2183] [Q&A]
Hi!

Occasionally (and correctly) I need a numeric cell to hold an Infinity value. If I use the Javascript value Number of Infinity, this causes problems with excel loading that datasheet (though it repairs it and assigns the string Infinity to the cell). I suspect the issue is to do with Excel's Infinity support, or lack thereof.

Thoughts on how should this be handled?

At present I explicitly test for Infinity and map to Number.MAX_VALUE on the assignment instead. This works in the current usage (just doing comparisons at present and not arithmetic), but isn't ideal. I could assign an error of #NUM (I think), but that also isn't ideal.

Simple code example is:
```
const ExcelJS     = require('exceljs')

const workbook = new ExcelJS.Workbook();
const worksheet = workbook.addWorksheet("Infinity Test")

worksheet.getRow(1).getCell(1).value = Infinity

workbook.xlsx.writeFile( 'infinity.xlsx' )
```

And Excel says:

![image](https://user-images.githubusercontent.com/91249741/208892200-344d223f-0143-4955-88ea-576ec5c264e6.png)

![image](https://user-images.githubusercontent.com/91249741/208892697-10aa653b-831b-4055-9d0d-b59a7b5d2b65.png)

And the log error is

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><logFileName>error068560_11.xml</logFileName><summary>Errors were detected in file 'https://synamedia-my.sharepoint.com/personal/kmurray_synamedia_com/Documents/src/aws-pricing-to-excel/infinity.xlsx'</summary><repairedRecords><repairedRecord>Repaired Records: Cell information from /xl/worksheets/sheet1.xml part</repairedRecord></repairedRecords></recoveryLog>
```

**Comment by SleekPanther** [comment]
We had this issue as well. I would have gone back to the source to see why the calculation was resulting in infinity, but the guy who fixed the issue used the workaround of sanitizing the output before sending to excel and converting the JavaScript Infinity to lowercase "infinity".

```
var SanitizeOutputForExcel = function(text) {
    // When the word "Infinity" get assigned to cell value, excel breaks so we use "infinity" which works.
    return (txt === "Infinity") ? "infinity" : txt;
};
```

This is called after parsing data from an html table so txt always is a string.
I haven't found any other answers and I guess it depends on if you want to display the word "infinity", the infinity symbol, or JavaScript's largest number.
